lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3086,9 +3086,14 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       return;
     }
 
-    TC.validateExtension(ED);
+    // Validate the nominal type declaration being extended.
+    TC.validateDecl(nominal);
+    // Don't bother computing the generic signature if the extended nominal
+    // type didn't pass validation so we don't crash.
```

Comment:
Do we have tests that crash without the isInvalid() check? It's unfortunate. We should fix that in a follow-up too.